Makefile

```diff
@@ -0,0 +1,323 @@
+releases-envsubst:
+	$(OPEN) $(ENVSUBST_RELEASES)
+
+KUBECTL_RELEASES := https://github.com/kubernetes/kubernetes/tags
+# Keep this in sync with KIND_K8s_VERSION
+KUBECTL_VERSION = 1.20.7
+KUBECTL_BIN := kubectl-$(KUBECTL_VERSION)-$(platform)-amd64
+KUBECTL_URL := https://storage.googleapis.com/kubernetes-release/release/v$(KUBECTL_VERSION)/bin/$(platform)/amd64/kubectl
+KUBECTL := $(LOCAL_BIN)/$(KUBECTL_BIN)
+$(KUBECTL): | $(CURL) $(LOCAL_BIN)
+	$(CURL) --progress-bar --fail --location --output $(KUBECTL) "$(KUBECTL_URL)"
+	touch $(KUBECTL)
+	chmod +x $(KUBECTL)
+	$(KUBECTL) version | grep $(KUBECTL_VERSION)
+	ln -sf $(KUBECTL) $(LOCAL_BIN)/kubectl
+.PHONY: kubectl
+kubectl: $(KUBECTL)
+.PHONY: releases-kubectl
+releases-kubectl:
+	$(OPEN) $(KUBECTL_RELEASES)
+K_CMD ?= apply
+# Dump all objects (do not apply) if DEBUG variable is set
+ifneq (,$(DEBUG))
+K_CMD = create --dry-run=client --output=yaml
+endif
+
+K9S_RELEASES := https://github.com/derailed/k9s/releases
+K9S_VERSION := 0.25.4
+K9S_BIN_DIR := $(LOCAL_BIN)/k9s-$(K9S_VERSION)-$(platform)-x86_64
+K9S_URL := $(K9S_RELEASES)/download/v$(K9S_VERSION)/k9s_$(platform)_x86_64.tar.gz
+K9S := $(K9S_BIN_DIR)/k9s
+$(K9S): | $(CURL) $(LOCAL_BIN) $(KUBECTL)
+	$(CURL) --progress-bar --fail --location --output $(K9S_BIN_DIR).tar.gz "$(K9S_URL)"
+	mkdir -p $(K9S_BIN_DIR) && tar zxf $(K9S_BIN_DIR).tar.gz -C $(K9S_BIN_DIR)
+	touch $(K9S)
+	chmod +x $(K9S)
+	$(K9S) version | grep $(K9S_VERSION)
+	ln -sf $(K9S) $(LOCAL_BIN)/k9s
+.PHONY: releases-k9s
+releases-k9s:
+	$(OPEN) $(K9S_RELEASES)
+.PHONY: k9s
+K9S_ARGS ?= --all-namespaces
+k9s: | $(KUBECONFIG) $(K9S) ## Terminal ncurses UI for K8s
+	$(K9S) $(K9S_ARGS)
+
+
+
+### TARGETS #
+#
+.DEFAULT_GOAL := help
+
+.PHONY: env
+env:: ## Configure shell env - eval "$(make env)" OR rm .env && make .env && source .env
+	@echo 'unalias m 2>/dev/null || true ; alias m=make'
+.env:
+	$(MAKE) --file $(lastword $(MAKEFILE_LIST)) --no-print-directory env SILENT="1>/dev/null 2>&1" > .env
+
+CURL ?= /usr/bin/curl
+$(CURL):
+	@which $(CURL) \
+	|| ( printf "$(RED)$(BOLD)$(CURL)$(RESET)$(RED) is missing, install $(BOLD)curl$(RESET)\n" ; exit 1)
+.PHONY: curl
+curl: $(CURL)
+
+HELP_TARGET_DEPTH ?= \#\#
+.PHONY: help
+help:
+	@awk -F':+ |$(HELP_TARGET_DEPTH)' '/^[^.][0-9a-zA-Z._%-]+:+.+$(HELP_TARGET_DEPTH).+$$/ { printf "\033[36m%-26s\033[0m %s\n", $$1, $$3 }' $(MAKEFILE_LIST) \
+	| sort
+
+define MAKE_TARGETS
+  awk -F':+' '/^[^.%\t_][0-9a-zA-Z._%-]*:+.*$$/ { printf "%s\n", $$1 }' $(MAKEFILE_LIST)
+endef
+
+define BASH_AUTOCOMPLETE
+  complete -W \"$$($(MAKE_TARGETS) | sort | uniq)\" make gmake m
+endef
+
+.PHONY: bash-autocomplete
+bash-autocomplete:
+	@echo "$(BASH_AUTOCOMPLETE)"
+env:: bash-autocomplete
+
+.PHONY: dep-update
+dep-update: ## Update any dependency
+	@printf "Update dep in go.mod by running e.g. $(BOLD)go get -d github.com/rabbitmq/messaging-topology-operator@v1.2.1$(RESET)\n" \
+	; read -rp " (press any key when done)" -n 1
+	$(CURDIR)/hack/update-deps.sh
```

Comment:
There were situations when I had to run `update-deps` before `update-codegen`, so this accounts for that.

Maybe running `update-deps` before & after `update-codegen` is the way to go?

Reply:
Yeah, if you're updating one of the RabbitMQ deps you'll need to run `update-deps` to update the dependency itself, and then `update-codegen` to update the generated clients, etc. We run `update-deps` at the end of codegen because in theory the codegen library might change and introduce new dependencies _in the generated code itself_.

This makes it hard to automate in a one-liner without always doing the most amount of redundant work possible. We could in theory have a dependency graph of this but that sounds hard.

Reply:
Updating deps twice seems like the simplest approach, even if it's not the fastest one. Your comment makes me realise that this still stands.